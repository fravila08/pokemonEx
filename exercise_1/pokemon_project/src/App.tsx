import { useEffect, useState } from 'react'
import reactLogo from './assets/react.svg'
import './App.css'
import { Container } from 'react-bootstrap'
import axios from 'axios'


export const getPokemon = async() =>{
  let response = await axios.get("https://pokeapi.co/api/v2/pokemon/ditto")
  return { 
    name : response.data['name'],
    img : response.data['sprites']['front_default']
  }
}

function App() {
  const [pokemon, setPokemon] = useState({})
  
  useEffect(()=>{
    const myPokemon =async () => {
      let pokeObject = await getPokemon()
      setPokemon(pokeObject)
    }
    myPokemon()
  },[])

  useEffect(()=>{
    console.log(pokemon)
  }, [pokemon])

  return (
    <Container className="App">
      <h1 id='titleHeader'>Pokemon API EX</h1>
    </Container>
  )
}

export default App
