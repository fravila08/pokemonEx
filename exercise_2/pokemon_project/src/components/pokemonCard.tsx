import { IPokemon } from "../App"
import Card from 'react-bootstrap/Card';

export interface PokemonCardProps{
    pokemon : IPokemon;
}

export const PokemonCard : React.FC<PokemonCardProps> = ({pokemon}) =>{

    return (
        <Card style={{ width: '18rem' }}>
            <Card.Img id="pokemonIMG" variant="top" src={pokemon.img} />
            <Card.Body>
                <Card.Title id="pokemonName">{pokemon.name}</Card.Title>
            </Card.Body>
        </Card>
    )
}