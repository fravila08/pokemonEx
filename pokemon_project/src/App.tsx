import { useState } from 'react'
import reactLogo from './assets/react.svg'
import './App.css'
import { Container } from 'react-bootstrap'

function App() {
  

  return (
    <Container className="App">
      <h1>Pokemon API EXAMPLE</h1>
    </Container>
  )
}

export default App
