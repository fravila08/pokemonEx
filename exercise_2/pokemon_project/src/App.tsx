import { useEffect, useState } from 'react'
import './App.css'
import { Container } from 'react-bootstrap'
import axios from 'axios'
import { PokemonCard } from './components/pokemonCard'

export interface IPokemon {
  name : string;
  img : string; 
}


export async function getPokemon(): Promise<IPokemon>{
  let response = await axios.get("https://pokeapi.co/api/v2/pokemon/ditto")
  return { 
    name : response.data['name'],
    img : response.data['sprites']['front_default']
  }
}

function App() {
  const [pokemon, setPokemon] = useState<IPokemon>({name:'none', img:'none'})
  const [show, setShow] = useState(false)
  useEffect(()=>{
    const myPokemon =async () => {
      let pokeObject = await getPokemon()
      setPokemon(pokeObject)
      setShow(true)
    }
    myPokemon()
  },[])

  useEffect(()=>{
    console.log(pokemon)
  }, [pokemon])

  return (
    <Container className="App">
      <h1 id='titleHeader'>Pokemon API EX</h1>
      {show ?
       <PokemonCard pokemon={pokemon} />
       :
       null
      }
    </Container>
  )
}

export default App
