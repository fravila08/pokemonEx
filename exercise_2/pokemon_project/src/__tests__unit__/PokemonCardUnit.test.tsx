import {describe, it , expect, } from "vitest";
import TestRenderer from 'react-test-renderer';
import { IPokemon } from "../App";
import { PokemonCard } from "../components/pokemonCard";


describe("PokemonCard", ()=>{
    it("Create snapshot of component", ()=>{
        let pokemon = {
            name : 'name',
            img : 'img'
        }
        const pokemonCard = TestRenderer.create(
            <PokemonCard pokemon={pokemon}/>
        )
        expect(pokemonCard).toMatchSnapshot();
    })
})