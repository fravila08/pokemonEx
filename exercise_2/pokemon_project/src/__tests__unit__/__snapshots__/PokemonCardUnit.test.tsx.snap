// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`PokemonCard > Create snapshot of component 1`] = `
<div
  className="card"
  style={
    {
      "width": "18rem",
    }
  }
>
  <img
    className="card-img-top"
    id="pokemonIMG"
    src="img"
  />
  <div
    className="card-body"
  >
    <div
      className="card-title h5"
      id="pokemonName"
    >
      name
    </div>
  </div>
</div>
`;
